nav2_costmap_2d/src/costmap_topic_collision_checker.cpp

```diff
@@ -92,25 +95,27 @@ double CostmapTopicCollisionChecker::scorePose(
 Footprint CostmapTopicCollisionChecker::getFootprint(const geometry_msgs::msg::Pose2D & pose)
 {
   Footprint footprint;
-  if (!footprint_sub_.getFootprint(footprint)) {
+  rclcpp::Time stamp(0L, clock_->get_clock_type());
+  if (!footprint_sub_.getFootprint(footprint, stamp)) {
     throw CollisionCheckerException("Current footprint not available.");
   }
 
   Footprint footprint_spec;
-  unorientFootprint(footprint, footprint_spec);
+  unorientFootprint(footprint, footprint_spec, stamp);
```

Comment:
You add stamp, which is just Time(0), and pass in this function to `getCurrentPose` where you added the stamp argument, which defaults to Time() if not provided. 

Which was the previous behavior without an argument at all, I don't think you ever use `getCurrentPose` which a non Time() value to have any change. Maybe worth reverting those changes? I see changes but no use of the features they would enable.

Reply:
Actually the stamp is retrieved inside `footprint_sub_.getFootprint(footprint, stamp)`. Yeah, the line 98 should be `rclcpp::Time stamp = clock_->now();` instead of Time(0) so that the footprint timeout check can be performed. That means, stamp is an in-out parameter of `getFootprint` method (in for timeout check, out for getting actual footprint timeout). I agree the usage of parameters of this method is not very intuitive and should be documented more in detail (I needed to study the code too to get a picture of how it actually works) or the API should be changed somehow (that's why I requested discussion in the issue report).